sale_order_revision/models/sale_order.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+# Copyright 2016 Serpent Consulting Services Pvt. Ltd.
+# Copyright 2018 Dreambits Technologies Pvt. Ltd. (<http://dreambits.in>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends('current_revision_id', 'old_revision_ids')
+    def _has_old_revisions(self):
+        for sale_order in self:
+            if sale_order.old_revision_ids:
+                sale_order.has_old_revisions = True
+            else:
+                sale_order.has_old_revisions = False
+
+    current_revision_id = fields.Many2one('sale.order',
+                                          'Current revision',
+                                          readonly=True,
+                                          copy=True)
+    old_revision_ids = fields.One2many('sale.order',
+                                       'current_revision_id',
+                                       'Old revisions',
+                                       readonly=True,
+                                       context={'active_test': False})
+    revision_number = fields.Integer('Revision',
+                                     copy=False,
+                                     default=0)
+    unrevisioned_name = fields.Char('Order Reference',
+                                    copy=True,
+                                    readonly=True)
+    active = fields.Boolean('Active',
+                            default=True)
+
+    has_old_revisions = fields.Boolean('Has old revisions',
+                                       compute=_has_old_revisions)
+
+    _sql_constraints = [
+        ('revision_unique',
+         'unique(unrevisioned_name, revision_number, company_id)',
+         'Order Reference and revision must be unique per Company.'),
+    ]
+
+    @api.returns('self', lambda value: value.id)
+    @api.multi
+    def copy(self, default=None):
+        if default is None:
+            default = {}
+        if self.env.context.get('new_sale_revision'):
```

Comment:
@devang-dreambits I think using context could be avoided as it can be done outside (calling another function).

Reply:
fixed in new commit